src/sentry/api/endpoints/release_thresholds/health_checks/is_crash_free_rate_healthy.py

```
@@ -0,0 +1,139 @@
+    return total
+
+
+def get_interval_indexes(intervals: list[str], start: datetime, end: datetime) -> Tuple[int, int]:
+    """
+    :param intervals: if timestamps from fetched sessions data
+    :param start: timestamp
+    :param end: timestamp
+
+    :return: If any subsection of the intervals fall within the given start/end, return the indexes of both the start and the end
+
+    TODO: raise if start/end not contained within intervals?
+    """
+    start_idx = len(intervals)
+    end_idx = 0
+
+    for idx, i in enumerate(intervals):
+        interval_date = parser.parse(i).replace(tzinfo=None)
+        if start <= interval_date < end:
+            if idx < start_idx:
+                start_idx = idx
+            if end_idx < idx:
+                end_idx = idx
+
+    return start_idx, end_idx
+
+
+def is_crash_free_rate_healthy_check(
+    ethreshold: EnrichedThreshold,
+    sessions_data: Dict[str, Any],
+    display: str,
+) -> Tuple[bool, float]:  # (is_healthy, metric_value)
+    """
+    Derives percent from crash total over total count
+
+    NOTE: Logic mirrors Frontend views/releases/list/releasesRequest::getCrashFreeRate
+    NOTE: if 'environment' not provided, will tally sums for _all_ sessions data as opposed to only sessions with environment of 'None'
+    """
+    field = "count_unique(user)" if display == CRASH_USERS_DISPLAY else "sum(session)"
+    release_version = ethreshold["release"]
+    project_id = ethreshold["project_id"]
+    environment = ethreshold["environment"]
+    intervals = sessions_data.get("intervals", [])
+    try:
+        start_idx, end_idx = get_interval_indexes(
+            intervals=intervals, start=ethreshold["start"], end=ethreshold["end"]
+        )
+        if start_idx > end_idx:
+            logger.warning("Threshold window not within provided session data")
+            raise ValueError("Threshold window not within provided session data")
+    except ValueError:
+        return False, 0
```

should we return something other than `0` to indicate an error occurred rather than assuming no crashes occurred? if the data passed in is somehow faulty is it the same as there actually being 0 crashes

---

That's a fair point 🤔 

Maybe -1?

---

maybe, but i'm not sure if the frontend needs to show something different for a `-1`

---

Yea - we can make that change

I'll update it to -1 for now